### ISSUE TITLE
Document functional differences between tf.stack and tf.parallel_stack

### ISSUE THREAD

**Issue by ahundt** [tensorflow/tensorflow#10036]
What if `tf.stack` had the API:

```
tf.stack(
    values,
    axis=0,
    name='stack',
    algorithm='sequential'
)
```

Options for algorithm could be `sequential`, `parallel` to start. This could apply broadly to many ops and ops like `parallel_stack` could be deprecated. 

Any ideas for a better parameter name than `algorithm`?

**Comment by skye** [comment]
@yuanbyu

**Comment by ebrevdo** [comment]
@ahundt `tf.parallel_stack` has strict limitations that `tf.stack` does not.  Not only do the shapes need to be known at graph build time for `tf.parallel_stack`, but this op ***does not support backprop*** (due to incompatibility between the in-place nature of its calculation and the functional requirements for enabling backprop on an op)!  In fact, this is more of a block temporary variable assign op than it is a stack op.  For this reason I prefer we keep this as an explicitly different operation.

**Comment by ebrevdo** [comment]
The fact that `parallel_stack` has no backprop support should be clearly documented, but it doesn't seem to be :(

**Comment by ebrevdo** [comment]
@alextp can you confirm?

**Comment by alextp** [comment]
Yes, it should be documented that it has no backprop support.

parallel_stack also has restrictions on what can be stacked (all shapes must be fully known I think)

**Comment by ebrevdo** [comment]
OK; so I'm going to rename this to "document functional differences between tf.parallel_stack and tf.stack."

**Comment by ebrevdo** [comment]
PRs welcome.

**Comment by ahundt** [comment]
That's definitely important information, thanks! 

Perhaps either (1) my suggested rename, or (2) a rename to something like `block_until_data_arrives(layout='parallel_stack')` could still be made? 

In addition to the clearer naming the limitation of the `parallel` vs `sequential` string option could of course still be documented.

Also, don't forget that I think there are still quite a few other pairs of ops which have similar names but differing functionality, I just chose `parallel_stack` as a good example!

**Comment by Androbin** [comment]
I submitted a PR to clarify things.
Tell me if you think the layout could direct the focus better.

**Comment by ahundt** [comment]
Despite the title change, I do still think specifying `algorithm='sequential'` and `algorithm='parallel'` params to `tf.stack()`, then deprecating `tf.parallel_stack()` would be cleaner. The description can of course specify all the useful information currently present in both docstrings, plus `'parallel' is faster for some use cases, but does not support backpropagation`.

Just figured I'd plug the idea one more time, no worries if you decide to stick to a docstring only change. :-)

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by wt-huang** [comment]
Closing as this is resolved
